src/modules/ethereum/transactions/gasPrices/optimistic/fee.ts

```diff
@@ -42,22 +47,41 @@ export async function createOptimisticFee({
   transaction: Transaction;
   getNonce: () => Promise<number>;
 }): Promise<OptimisticFee | null> {
-  const { l1, l2, fixed_overhead, dynamic_overhead } = optimisticGasPriceInfo;
-  if (
-    l1 == null ||
-    l2 == null ||
-    fixed_overhead == null ||
-    dynamic_overhead == null
-  ) {
+  const { l1, fixed_overhead, dynamic_overhead } = optimisticGasPriceInfo;
+  if (l1 == null || fixed_overhead == null || dynamic_overhead == null) {
     return null;
   }
-  const { gas, to = '0x', value = '0x0', data = '0x' } = transaction;
+  const gas = getGas(transaction) as string;
+  const {
+    to = '0x',
+    value = '0x0',
+    data = '0x',
+    gasPrice,
+    maxFeePerGas,
+    maxPriorityFeePerGas,
+  } = transaction;
   if (gas == null) {
     return null;
   }
 
   const nonce = await getNonce();
-  const encoded_tx_data = rlpEncode([nonce, l2, gas, to, value, data]);
+  const encoded_tx_data =
+    maxPriorityFeePerGas != null && maxFeePerGas != null
+      ? rlpEncode([
+          nonce,
+          gas,
+          to,
+          value,
+          data,
+          maxPriorityFeePerGas,
+          maxFeePerGas,
+        ])
+      : gasPrice != null
+      ? rlpEncode([nonce, gasPrice, gas, to, value, data])
```

Comment:
is this order of params for real a part of the optimistic algorithm? :D

Reply:
Yes :((

Reply:
I've doublechecked and seems like in the final version of [EIP1559](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1559.md#specification) they decided to use more adequate order of perameters.

Reply:
So this one is better? `The [EIP-2718](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2718.md) TransactionPayload for this transaction is rlp([chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, destination, amount, data, access_list, signature_y_parity, signature_r, signature_s]).`